inmoose/consensus_clustering/consensus_clustering.py

```diff
@@ -0,0 +1,454 @@
+            Array of the item consensus (nb_items * k)
+        """
+        assert self.min_k <= k <= self.max_k, "Number of clusters must be between min_k and max_k"
+        items_consensus = np.zeros((self.consensus_matrices[k-self.min_k].shape[0], k))
+
+        clusters, sizes = np.unique(prediction, return_counts=True)
+
+        for id in range(items_consensus.shape[0]):
+            for clust, siz in zip(clusters, sizes):
+
+                clust_elem = np.where(prediction==clust)[0]
+                if id in clust_elem:
+                    cols = clust_elem[clust_elem!=id]
+                    items_consensus[id, clust]= np.sum(self.consensus_matrices[k-self.min_k, id, cols])/(siz-1)
+                else:
+                    items_consensus[id, clust]= np.sum(self.consensus_matrices[k-self.min_k, id, clust_elem])/siz
+        return items_consensus
+
+    def predict(self, k):
+        """
+        Predicts clusters on the consensus matrix, for k clusters using the consensus matrix
+
+        Arguments
+        -----------
+        k: int
+            Number of clusters
+
+        Returns
+        -----------
+        predicted cluster for k clusters
+        """
+        assert self.consensus_matrices is not None, "First compute consensus clustering"
+        assert self.min_k <= k <= self.max_k, "Number of clusters must be between min_k and max_k"
+        return self.cluster_(n_clusters=k).fit_predict(
+            1-self.consensus_matrices[k-self.min_k])
+
+    def predict_data(self, data):
+        """
+        Predicts clusters on the data, for best found cluster number
+
+        Arguments
+        -----------
+        data: ndarray
+            input matrix (samples * attributes)
+
+        Returns
+        -----------
+        predicted cluster for best number of clusters
+        """
+        assert self.consensus_matrices is not None, "First compute consensus clustering"
+        return self.cluster_(n_clusters=self.bestK).fit_predict(
+            data)
+
+    def plot_clustermap(self, k, saving_path, col_color=None):
+        """
+        Plot and save clustermap
+
+        Arguments
+        -----------
+        k: int
+            The number of cluster to plot the clustermap for
+        saving_path: str
+            Path to file where to store the result plot
+        col_color: array[str]
+            array of color to annotate the clustermap.
+            NOTE: the colors must be ordered in the same way as the initial data matrix.
+
+        """
+        assert self.consensus_matrices is not None, "First compute consensus clustering"
+        assert self.min_k <= k <= self.max_k, "Number of clusters must be between min_k and max_k"
+        cluster_map = sn.clustermap(self.consensus_matrices[k-self.min_k], method="ward", metric="euclidean",
+                cmap=sn.color_palette("Blues", as_cmap=True),
+                yticklabels=False,xticklabels=False, figsize=(10,10), col_colors=col_color)
```

Comment:
Are you sure that an array of string is enough here if we want to stack multiple data elements?
In clustermap example, it looks more like a series (or a dataframe for multiple info) @soleneweill